src/Compilers/Core/Portable/PublicAPI.Unshipped.txt

```diff
@@ -190,6 +190,43 @@ Microsoft.CodeAnalysis.OperationKind.VariableDeclarationStatement = 3 -> Microso
 Microsoft.CodeAnalysis.OperationKind.WithStatement = 82 -> Microsoft.CodeAnalysis.OperationKind
 Microsoft.CodeAnalysis.OperationKind.YieldBreakStatement = 12 -> Microsoft.CodeAnalysis.OperationKind
 Microsoft.CodeAnalysis.OperationKind.YieldReturnStatement = 16 -> Microsoft.CodeAnalysis.OperationKind
+Microsoft.CodeAnalysis.Options.IOption
+Microsoft.CodeAnalysis.Options.IOption.DefaultValue.get -> object
+Microsoft.CodeAnalysis.Options.IOption.Feature.get -> string
+Microsoft.CodeAnalysis.Options.IOption.IsPerLanguage.get -> bool
+Microsoft.CodeAnalysis.Options.IOption.Name.get -> string
+Microsoft.CodeAnalysis.Options.IOption.StorageLocations.get -> System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Options.OptionStorageLocation>
+Microsoft.CodeAnalysis.Options.IOption.Type.get -> System.Type
+Microsoft.CodeAnalysis.Options.Option<T>
+Microsoft.CodeAnalysis.Options.Option<T>.DefaultValue.get -> T
+Microsoft.CodeAnalysis.Options.Option<T>.Feature.get -> string
+Microsoft.CodeAnalysis.Options.Option<T>.Name.get -> string
+Microsoft.CodeAnalysis.Options.Option<T>.Option(string feature, string name) -> void
+Microsoft.CodeAnalysis.Options.Option<T>.Option(string feature, string name, T defaultValue) -> void
+Microsoft.CodeAnalysis.Options.Option<T>.Option(string feature, string name, T defaultValue, params Microsoft.CodeAnalysis.Options.OptionStorageLocation[] storageLocations) -> void
+Microsoft.CodeAnalysis.Options.Option<T>.StorageLocations.get -> System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.Options.OptionStorageLocation>
+Microsoft.CodeAnalysis.Options.Option<T>.Type.get -> System.Type
+Microsoft.CodeAnalysis.Options.OptionKey
+Microsoft.CodeAnalysis.Options.OptionKey.Equals(Microsoft.CodeAnalysis.Options.OptionKey other) -> bool
+Microsoft.CodeAnalysis.Options.OptionKey.Language.get -> string
+Microsoft.CodeAnalysis.Options.OptionKey.Option.get -> Microsoft.CodeAnalysis.Options.IOption
+Microsoft.CodeAnalysis.Options.OptionKey.OptionKey(Microsoft.CodeAnalysis.Options.IOption option, string language = null) -> void
+Microsoft.CodeAnalysis.Options.OptionSet
+Microsoft.CodeAnalysis.Options.OptionSet.GetOption<T>(Microsoft.CodeAnalysis.Options.Option<T> option) -> T
+Microsoft.CodeAnalysis.Options.OptionSet.GetOption<T>(Microsoft.CodeAnalysis.Options.PerLanguageOption<T> option, string language) -> T
+Microsoft.CodeAnalysis.Options.OptionSet.OptionSet() -> void
+Microsoft.CodeAnalysis.Options.OptionSet.WithChangedOption<T>(Microsoft.CodeAnalysis.Options.Option<T> option, T value) -> Microsoft.CodeAnalysis.Options.OptionSet
+Microsoft.CodeAnalysis.Options.OptionSet.WithChangedOption<T>(Microsoft.CodeAnalysis.Options.PerLanguageOption<T> option, string language, T value) -> Microsoft.CodeAnalysis.Options.OptionSet
+Microsoft.CodeAnalysis.Options.OptionStorageLocation
```

Comment:
❗️ I don't believe this type should be part of the public API. However, I didn't make changes to the way it was implemented at this time.

@jasonmalinowski If we don't find a way to separate option serialization from the options themselves (which #12593 was trying to avoid), we're going to end up either unable to expose options accurately in the compiler layer, or the compiler layer will end up specifying Visual Studio IDE-specific serialization behavior (e.g. `RoamingProfileStorageLocation`).